exporter/views.py

```diff
@@ -10,8 +11,33 @@
 from django.views.generic import TemplateView
 from tabular_export.core import export_to_csv_response, flatten_queryset
 
-from concordia.models import Asset, Campaign, Transcription
-from concordia.storage import ASSET_STORAGE
+from concordia.models import Asset, Transcription
+
+
+def get_latest_transcription_data(campaign_slug):
+    latest_trans_subquery = (
+        Transcription.objects.filter(asset=OuterRef("pk"))
+        .order_by("-pk")
+        .values("text")
+    )
+    assets = Asset.objects.annotate(
+        latest_transcription=Subquery(latest_trans_subquery[:1])
+    )
+    assets = assets.filter(item__project__campaign__slug=campaign_slug)
+    return assets
+
+
+def get_original_asset_id(download_url):
+    """
+    Extract the bit from the download url
+    that identifies this image uniquely on loc.gov
+    """
+    if download_url.startswith("http://tile.loc.gov/"):
+        pattern = r"/service:([A-Za-z0-9:]*)/"
+        asset_id = re.search(pattern, download_url)
```

Comment:
Shouldn't we handle the case where this fails with at least an `assert asset_id`?

Reply:
Yes, this part is still under development. I need to check all the download_urls for all the assets to see which ones if any did not come from the tile server using this URL convention.